ui/store/actions.ts

```diff
@@ -7107,34 +7107,14 @@ export async function endBackgroundTrace(request: EndTraceRequest) {
   ]);
 }
 
-/**
- * Apply the state patches from the background.
- * Intentionally not using immer as a temporary measure to avoid
- * freezing the resulting state and requiring further fixes
- * to remove direct state mutations.
- *
- * @param oldState - The current state.
- * @param patches - The patches to apply.
- * Only supports 'replace' operations with a single path element.
- * @returns The new state.
- */
 function applyPatches(
   oldState: Record<string, unknown>,
   patches: Patch[],
 ): Record<string, unknown> {
-  const newState = { ...oldState };
-
-  for (const patch of patches) {
-    const { op, path, value } = patch;
-
-    if (op === 'replace') {
-      newState[path[0]] = value;
-    } else {
-      throw new Error(`Unsupported patch operation: ${op}`);
-    }
-  }
+  const immer = new Immer();
+  immer.setAutoFreeze(false);
```

Comment:
Hmm. It's unfortunate that we're not freezing here, though I know it wasn't frozen before either, so no need to change that here. 

Just wondering, did you encounter an error related to this being frozen (e.g. was something mutating this state)? Or was auto-freeze disabled as a precaution?

Reply:
Oh nevermind, I took a closer look at the commit history here and it answered the question. Yes, we definitely have state mutations.

Reply:
Unfortunately yes, would love to freeze also but didn't want to add further scope and risk to this PR.